diff --git a/src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/ToDownloadProcessor.scala b/src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/ToDownloadProcessor.scala
index e0ba076f8..41b9d98b8 100644
--- a/src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/ToDownloadProcessor.scala
+++ b/src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/ToDownloadProcessor.scala
@@ -90,7 +90,8 @@ trait ToDownloadProcessor
       case Some(fb) =>
         // when blockchain is about to be synced,
         // download children blocks of last 100 full blocks applied to the best chain, to get block sections from forks
-        val minHeight = Math.max(1, fb.header.height - 100)
+        // (but not below minimal full block height, as there could be no headers below it, e.g. after NiPoPoW bootstrap)
+        val minHeight = Math.max(minimalFullBlockHeight, fb.header.height - 100)
         continuation(minHeight, Map.empty, maxHeight = Int.MaxValue)
       case None if (nodeSettings.utxoSettings.utxoBootstrap && !isUtxoSnapshotApplied) =>
         // if bootstrapping with UTXO set snapshot is chosen, and no snapshot applied yet, ask peers for snapshots
